feature-based-molecular-networking/tools/feature-based-molecular-networking/scripts/progenesis_formatter.py

```diff
@@ -147,17 +152,39 @@ def convert_mgf(input_msp, output_mgf, compound_to_scan_mapping):
 
             read_name = True
             scan = (compound_to_scan_mapping[compound_name])
-        elif line.startswith("PrecursorMZ:"):
-            precursor_mz = (line.rstrip().replace("PrecursorMZ: ", ""))
-        elif line.startswith("Charge:"):
-            charge = (line.rstrip().replace("Charge: ", ""))
-        elif len(line.rstrip()) == 0 and read_name == True:
+
+        # This gets the precursor ion mass
+        if line.startswith("PrecursorMZ:"):
+            precursor_mz = (line.rstrip().replace("PrecursorMZ: ", "PEPMASS="))
+        # This gets the charge (either positive or negative) from the Charge header
+        if line.startswith("Charge:"):
+            charge = []
+            charge = (line.rstrip().replace("Charge: ", "CHARGE="))
+            precursor_type = []
+        # When there is an adduct provided, use the adduct to deduce the charge, because there is no Charge header anymore.
+        elif line.startswith("Precursor_type:"):
+            precursor_type = []
+            precursor_type = line.rstrip().replace("Precursor_type: ", "ION=")
+            charge = []
+        if len(line.rstrip()) == 0 and read_name == True:
             read_name = False
 
             output_filename.write("BEGIN IONS\n")
             output_filename.write("SCANS=%s\n" % (compound_to_scan_mapping[compound_name]))
+            output_filename.write("FEATURE_ID=%s\n" % (compound_to_scan_mapping[compound_name]))
```

Comment:
is it FEATURE_ID or FEATUREID ? I thought no underscore in MZMine2 output.

Reply:
FEATURE_ID